baybe/acquisition/acqfs.py

```diff
@@ -1,11 +1,122 @@
+
+    Cannot be used if `sampling_fraction` is not `None`."""
+
+    sampling_fraction: float | None = field(
+        converter=optional_c(float),
+        validator=optional_v([gt(0.0), le(1.0)]),
+    )
+    """Fraction of data sampled for integrating the posterior.
+
+    Cannot be used if `sampling_n_points` is not `None`."""
+
+    sampling_method: DiscreteSamplingMethod = field(
+        converter=DiscreteSamplingMethod, default=DiscreteSamplingMethod.Random
+    )
+    """Sampling strategy used for integrating the posterior."""
+
+    @sampling_fraction.default
+    def _default_sampling_fraction(self) -> float | None:
+        """If no sampling quantities are provided, use all points by default."""
+        return 1.0 if self.sampling_n_points is None else None
+
+    @sampling_fraction.validator
+    def _validate_sampling_fraction(self, attr, value) -> None:
+        """If both sampling quantities are specified, raise an error."""
+        if None not in (self.sampling_fraction, self.sampling_n_points):
+            raise ValueError(
+                f"For '{self.__class__.__name__}', the attributes '{attr.name}' and "
+                f"'{fields(self.__class__).sampling_n_points.name}' cannot "
+                f"be specified at the same time."
+            )
+
+    def get_integration_points(self, searchspace: SearchSpace) -> pd.DataFrame:
+        """Sample points from a search space for integration purposes.
+
+        Sampling of the discrete part can be controlled via 'sampling_method', but
+        sampling of the continuous part will always be random.
+
+        Args:
+            searchspace: The searchspace from which to sample integration points.
+
+        Returns:
+            The sampled data points.
+
+        Raises:
+            ValueError: If the search space is purely continuous and
+                'sampling_n_points' was not provided.
+        """
+        sampled_parts: list[pd.DataFrame] = []
+        n_candidates: int | None = None
+
+        # Discrete part
+        if not searchspace.discrete.is_empty:
+            candidates_discrete = searchspace.discrete.comp_rep
+            n_candidates = self.sampling_n_points or math.ceil(
+                self.sampling_fraction * len(candidates_discrete)  # type: ignore[operator]
+            )
+
+            sampled_disc = sample_numerical_df(
+                candidates_discrete, n_candidates, method=self.sampling_method
+            )
+
+            sampled_parts.append(sampled_disc)
+
+        # Continuous part
+        if not searchspace.continuous.is_empty:
+            # If a discrete part has resulted in a particular choice for n_candidates,
+            # take it. Otherwise, use the user specified number of points.
+            if (n_candidates := n_candidates or self.sampling_n_points) is None:
+                raise ValueError(
+                    f"'sampling_n_points' must be provided for '{self.__class__}' when "
```

Comment:
Also here better to access name of `sampling_n_points` programatically

Reply:
included

Reply:
actually, do you have a suggestion on how to? This object has no `__name__`

Reply:
you can go via `attrs.fields(<cls>).<attr>.name`

Reply:
ty